plugin/cpp/src/dist/cpp.js

```diff
@@ -0,0 +1,247 @@
+  var CppCancelFunction = null;
+
+  var CppSh = {
+    pluginName: PLUGIN_NAME,
+    cmMode: 'text/x-c++src',
+    background: '#4757B8',
+    bgColor: '#4757B8',
+    fgColor: '#FFFFFF',
+    borderColor: '',
+    shortName: 'C',
+    newShell: function(shellId, cb) {
+      if (!shellId) {
+        shellId = '';
+      }
+      bkHelper.httpPost(bkHelper.serverUrl(serviceBase + '/rest/cpp/getShell'), {
+        shellId: shellId,
+        sessionId: bkHelper.getSessionId()})
+      .success(cb)
+      .error(function() {
+        console.log('failed to create shell', arguments);
+      });
+    },
+    evaluate: function(code, modelOutput, refreshObj, cellId) {
+      var deferred = Q.defer();
+
+      if (CppCancelFunction) {
+        deferred.reject('An evaluation is already in progress');
+        return deferred.promise;
+      }
+
+      var self = this;
+      bkHelper.setupProgressOutput(modelOutput);
+      $.ajax({
+        type: 'POST',
+        datatype: 'json',
+        url: bkHelper.serverUrl(serviceBase + '/rest/cpp/evaluate'),
+        data: {
+          shellId: self.settings.shellID,
+          cellId: cellId,
+          code: code
+        }
+      }).done(function(ret) {
+        CppCancelFunction = function() {
+          $.ajax({
+            type: 'POST',
+            datatype: 'json',
+            url: bkHelper.serverUrl(serviceBase + '/rest/cpp/cancelExecution'),
+            data: {shellId: self.settings.shellID}
+          }).done(function(ret) {
+            console.log('done cancelExecution', ret);
+          });
+          bkHelper.setupCancellingOutput(modelOutput);
+        }
+        var onEvalStatusUpdate = function(evaluation) {
+          if (bkHelper.receiveEvaluationUpdate(modelOutput, evaluation, PLUGIN_NAME, self.settings.shellID)) {
+            cometdUtil.unsubscribe(evaluation.update_id);
+            CppCancelFunction = null;
+            if (evaluation.status === 'ERROR') {
+              deferred.reject(evaluation.payload);
+            } else {
+              deferred.resolve(evaluation.payload);
+            }
+          }
+          if (refreshObj !== undefined) {
+            refreshObj.outputRefreshed();
+          } else {
+            bkHelper.refreshRootScope();
+          }
+        };
+        onEvalStatusUpdate(ret);
+        if (ret.update_id) {
+          cometdUtil.subscribe(ret.update_id, onEvalStatusUpdate);
+        }
+      });
+      return deferred.promise;
+    },
+    interrupt: function() {
+      this.cancelExecution();
+    },
+    cancelExecution: function() {
+      if (CppCancelFunction) {
+        CppCancelFunction();
+      }
+    },
+    resetEnvironment: function() {
+      $.ajax({
+        type: 'POST',
+        datatype: 'json',
+        url: bkHelper.serverUrl(serviceBase + '/rest/cpp/resetEnvironment'),
+        data: {shellId: this.settings.shellID}
+      }).done(function(ret) {
+        console.log('done resetEnvironment', ret);
```

Comment:
Huh. So all that stuff is a clone of the javash plugin, so I do not know. @scottdraves do we want these logs?